data/lua/carryover_gold.lua

```diff
@@ -0,0 +1,248 @@
+		res = res .. "<b>" .. _("Turns finished early: ") .. info.turns_left .. "</b>\n"
+	end
+	if info.finishing_bonus_per_turn ~= nil then
+		res = res .. "<small>" .. _("Early finish bonus: ") .. info.finishing_bonus_per_turn .. _(" per turn") .. "</small>\n"
+	end
+	res = res .. "<small>" .. _("Total bonus: ") .. info.finishing_bonus .. "</small>"
+
+	return res
+end
+
+---@param side side
+---@param info table
+---@return string
+function carryover.total_gold_message(side, info)
+	return "<small>" .. _("Total gold: ") .. carryover.half_signed_value(side.gold + info.finishing_bonus) .. "</small>"
+end
+
+---@param side side
+---@param info table
+---@return string
+function carryover.percentage_remaining_message(side, info)
+	return "<small>" .. _("Carryover percentage: ") .. side.carryover_percentage .. "</small>"
+end
+--- returns the last part of the gold report for a single side, describing how much gold is added in the next scenario.
+---@param side side
+---@param info table
+---@return string
+function carryover.carryover_mesage(side, info)
+	local goldmsg = ""
+	if side.carryover_add then
+		goldmsg = goldmsg .. "<big><b>" .. _("Bonus gold: ") .. carryover.half_signed_value(side.carryover_gold) .. "</b></big>"
+	else
+		goldmsg = goldmsg .. "<big><b>" .. _("Retained gold: ") .. carryover.half_signed_value(side.carryover_gold) .. "</b></big>"
+	end
+	goldmsg = goldmsg  .. "\n"
+	-- Note that both strings are the same in English, but some languages will
+	-- want to translate them differently.
+	if side.carryover_add then
+		if side.carryover_gold > 0 then
+			goldmsg = goldmsg .. _("You will start the next scenario with $gold on top of the defined minimum starting gold.",
+				"You will start the next scenario with $gold on top of the defined minimum starting gold.", side.carryover_gold)
+
+		else
+			goldmsg = goldmsg .. _("You will start the next scenario with the defined minimum starting gold.")
+		end
+	else
+		goldmsg = goldmsg .. _(
+			"You will start the next scenario with $gold or its defined minimum starting gold, whichever is higher.",
+			"You will start the next scenario with $gold or its defined minimum starting gold, whichever is higher.",
+			side.carryover_gold)
+	end
+
+	-- xgettext:no-c-format
+	return goldmsg:vformat{ gold = tostring(side.carryover_gold) }
+end
+
+--- returns the default gold report for a single side.
+---@param side side
+---@param info table
+---@return string
+function carryover.get_side_gold_report(side, info)
+
+	local report = ""
+
+	report = report .. carryover.remaining_gold_message(side, info)
+
+	if side.carryover_bonus ~= 0 then
+		report = report .. "\n\n" .. carryover.bonus_gold_message(side, info) .. "\n" .. carryover.total_gold_message(side, info)
+	end
+
+	if side.carryover_gold > 0 then
+		report = report .. "\n" .. carryover.percentage_remaining_message(side, info)
+	end
+
+
+	report = report .. "\n\n" .. carryover.carryover_mesage(side, info)
+	return report
+end
+
+---@return boolean
```

Comment:
Please add a comment indicating what this function does. I realize it may seem obvious from the name, but I'd like to make it explicit too.